lib/compat/wordpress-6.1/block-template-utils.php

```diff
@@ -195,34 +211,58 @@ function gutenberg_get_block_template( $id, $template_type = 'wp_template' ) {
 	if ( count( $parts ) < 2 ) {
 		return null;
 	}
-	list( $theme, $slug ) = $parts;
-	$wp_query_args        = array(
-		'post_name__in'  => array( $slug ),
-		'post_type'      => $template_type,
-		'post_status'    => array( 'auto-draft', 'draft', 'publish', 'trash' ),
-		'posts_per_page' => 1,
-		'no_found_rows'  => true,
-		'tax_query'      => array(
-			array(
-				'taxonomy' => 'wp_theme',
-				'field'    => 'name',
-				'terms'    => $theme,
-			),
-		),
-	);
-	$template_query       = new WP_Query( $wp_query_args );
-	$posts                = $template_query->posts;
+	list( , $slug ) = $parts;
 
-	if ( count( $posts ) > 0 ) {
-		$template = gutenberg_build_block_template_result_from_post( $posts[0] );
+	$templates      = gutenberg_get_template_slugs( $slug );
+	$block_template = resolve_block_template( $slug, $templates, '' );
+	if ( ! $block_template ) {
+		$block_template = resolve_block_template( 'index', array(), '' );
+	}
+	// This might give us a fallback template with a different ID,
+	// so we have to override it to make sure it's correct.
+	$block_template->id     = $id;
+	$block_template->slug   = $slug;
+	$default_template_types = get_default_block_template_types();
+
+	$slug_parts = explode( '-', $slug, 3 );
+	if ( count( $slug_parts ) > 1 ) {
+		if ( 'single' === $slug_parts [0] ) {
+			// Get CPT labels
+			$post_type = get_post_type_object( $slug_parts[1] );
+			$labels    = $post_type->labels;
 
-		if ( ! is_wp_error( $template ) ) {
-			return $template;
+			if ( count( $slug_parts ) > 2 ) {
+				// Now we look for the CPT with slug as defined in $slug_parts[2]
+				$post                  = get_page_by_path( $slug_parts[2], OBJECT, $slug_parts[1] );
```

Comment:
Also can't we just do this? 
```suggestion
				$post                  = get_page_by_path( $slug_parts[2], OBJECT, $template_type );
```

Reply:
I'm afraid that won't work -- `$template_type` will be something like `single`, `category`, or even `404`.

If it's `single`, then `$slug_parts[1]` will be the _post_ type that the template is for.

(`$template_type` OTOH will be reflected in `$slug_parts[0]`.)